### PR TITLE
Allow to request all nodes from discovery with monitoring data

### PR DESCRIPTION
**Pull request by soffokl**
Closes: #1959

### CODE REVIEW

File: config/flags_node.go [hunk 1]
```diff
@@ -170,6 +170,12 @@ var (
 		Usage: "Marks vendor (distributor) of the node for collecting statistics. " +
 			"3rd party vendors may use their own identifier here.",
 	}
+	// FlagUIEnable enables built-in web UI for node.
+	FlagIncludeFailedNodes = cli.BoolFlag{
```
Comment (Waldz) [inline config/flags_node.go:174]: Don't introduce this flag, all filtering "configuration" - is passing Tequilapi parameters. I see you used it only method `mAPI.Proposals()` which is not used.

File: core/discovery/apidiscovery/repository.go [hunk 2]
```diff
@@ -41,6 +42,7 @@ func (a *apiRepository) Proposal(id market.ProposalID) (*market.ServiceProposal,
 		NodeKey:         id.ProviderID,
 		ServiceType:     id.ServiceType,
 		AccessPolicyAll: true,
+		IncludeFailed:   config.GetBool(config.FlagIncludeFailedNodes),
```
[the remark this answers is not in the document]
Comment (soffokl) [reply core/discovery/apidiscovery/repository.go]: Removed it.

File: config/flags_node.go (hunk 1 above)
Comment (soffokl) [reply config/flags_node.go]: Removed it.

File: tequilapi/endpoints/proposals.go [hunk 3]
```diff
@@ -226,6 +237,7 @@ func (pe *proposalsEndpoint) List(resp http.ResponseWriter, req *http.Request, p
 		proposalsRes.Proposals = append(proposalsRes.Proposals, proposalToRes(p))
 	}
 
+	fetchConnectCounts := req.URL.Query().Get("fetch_connect_counts")
```
Comment (Waldz) [inline tequilapi/endpoints/proposals.go:240]: Probably now `fetch_metrics` is better name for that it does. Or do do granular fetch approach with query:
`&fetch_metric[connect_counts]=false&fetch_metric[monitoring_failed]=true`

File: tequilapi/endpoints/proposals.go [hunk 4]
```diff
@@ -214,6 +224,7 @@ func (pe *proposalsEndpoint) List(resp http.ResponseWriter, req *http.Request, p
 		LowerTimePriceBound: lowerTimePriceBound,
 		UpperTimePriceBound: upperTimePriceBound,
 		ExcludeUnsupported:  true,
+		IncludeFailed:       req.URL.Query().Get("include_failed") == "true",
```
Comment (Waldz) [inline tequilapi/endpoints/proposals.go:227]: `monitoring_failed=true` or `metrics.monitoring_failed=true` would be better name here as:
- as you would not leak API naming to Tequilapi contract
- it's Tequilapi endpoint , so alligned to Tequilapi proposal DTO

File: tequilapi/endpoints/proposals.go [hunk 5]
```diff
@@ -57,8 +57,20 @@ type serviceDefinitionRes struct {
...
+type qualityMetricsRes struct {
+	ProviderID  string `json:"provider_id"`
+	ServiceType string `json:"service_type"`
+	metricsRes
+}
+
 type metricsRes struct {
-	ConnectCount quality.ConnectCount `json:"connect_count"`
+	ConnectCount     quality.ConnectCount `json:"connect_count"`
+	MonitoringFailed bool                 `json:"monitoring_failed"`
```
Comment (Waldz) [inline tequilapi/endpoints/proposals.go:73]: BTW, are you introducing this to Tequilapi `client.go` somehow? We are migrating to same DTOs there at `contract` package.

File: tequilapi/endpoints/proposals.go (hunk 3 above)
Comment (soffokl) [reply tequilapi/endpoints/proposals.go]: Renamed.

File: tequilapi/endpoints/proposals.go (hunk 4 above)
Comment (soffokl) [reply tequilapi/endpoints/proposals.go]: Renamed.

File: tequilapi/endpoints/proposals.go (hunk 5 above)
Comment (soffokl) [reply tequilapi/endpoints/proposals.go]: It is in the `contract` package now.

**Pull request merged**